apps/viewer/src/app.tsx

```diff
@@ -69,17 +71,32 @@ const App = () => {
     }
   }, [remoteTrackSources.size]);
 
+  // Change main stream
   useEffect(() => {
     if (mainSourceId) {
-      // Change main stream
       projectToMainStream(mainSourceId).then(() => {
         // Reset quality
-        setSourceQuality(mainSourceId, { streamQuality: 'Auto' });
+        setSourceQuality(mainSourceId);
       });
     }
   }, [mainSourceId]);
 
-  const mainSource = mainSourceId !== undefined ? remoteTrackSources.get(mainSourceId) : undefined;
+  // Reset main stream when layers change
+  useEffect(() => {
+    if (!mainSource) {
+      return;
+    }
+
+    const { quality, sourceId, streamQualityOptions } = mainSource;
+    const streamQualities = streamQualityOptions.map(({ streamQuality }) => streamQuality);
+
+    if (!quality || !streamQualities.includes(quality)) {
+      // Must reproject before resetting quality
+      projectToMainStream(sourceId).then(() => {
```

Comment:
Could you please ask and confirm with Millicast team, do we need to re-project? Otherwise, there might be a hidden issue we don't know in our hook.